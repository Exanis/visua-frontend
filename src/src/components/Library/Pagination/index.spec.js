import React from 'react';
import { shallow } from 'enzyme';

import { PaginationBase } from './index';

describe('<Pagination />', () => {
    const defaultProps = {
        count: 145,
        onClick: () => {},
        classes: {
            bordered: 'style-bordered',
            horizontal: 'style-horizontal',
            active: 'style-active'
        }
    };

    it('should render Pagination properly', () => {
        const rendererComponent = shallow(<PaginationBase {...defaultProps} />)

        expect(rendererComponent).toMatchSnapshot();
    });

    it('should render Pagination properly when there is no items', () => {
        const props = {
            ...defaultProps,
            count: 0
        };
        const rendererComponent = shallow(<PaginationBase {...props} />)

        expect(rendererComponent).toMatchSnapshot();
    });

    it('should render Pagination properly when pagesize is specified', () => {
        const props = {
            ...defaultProps,
            pageSize: 3
        };
        const rendererComponent = shallow(<PaginationBase {...props} />)

        expect(rendererComponent).toMatchSnapshot();
    });

    it('should render Pagination properly when there is only one page', () => {
        const props = {
            ...defaultProps,
            count: 29
        };
        const rendererComponent = shallow(<PaginationBase {...props} />)

        rendererComponent.instance().page = 2;
        rendererComponent.update();
        expect(rendererComponent).toMatchSnapshot();
    });
});