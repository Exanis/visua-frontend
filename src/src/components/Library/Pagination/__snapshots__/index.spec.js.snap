// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render Pagination properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaginationBase
    classes={
        Object {
            "active": "style-active",
            "bordered": "style-bordered",
            "horizontal": "style-horizontal",
          }
    }
    count={145}
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="style-horizontal"
>
          <WithStyles(Typography)
                    className="style-bordered style-active"
                    onClick={[Function]}
                    variant="button"
          >
                    1
          </WithStyles(Typography)>
</div>,
        <div
          className="style-horizontal"
>
          <WithStyles(Typography)
                    className="style-bordered"
                    onClick={[Function]}
                    variant="button"
          >
                    2
          </WithStyles(Typography)>
</div>,
        <div
          className="style-horizontal"
>
          <WithStyles(Typography)
                    variant="body2"
          >
                    ...
          </WithStyles(Typography)>
          <WithStyles(Typography)
                    className="style-bordered"
                    onClick={[Function]}
                    variant="button"
          >
                    10
          </WithStyles(Typography)>
</div>,
      ],
      "className": "style-horizontal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <WithStyles(Typography)
              className="style-bordered style-active"
              onClick={[Function]}
              variant="button"
>
              1
</WithStyles(Typography)>,
          ],
          "className": "style-horizontal",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 1,
              "className": "style-bordered style-active",
              "onClick": [Function],
              "variant": "button",
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <WithStyles(Typography)
              className="style-bordered"
              onClick={[Function]}
              variant="button"
>
              2
</WithStyles(Typography)>,
          ],
          "className": "style-horizontal",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 2,
              "className": "style-bordered",
              "onClick": [Function],
              "variant": "button",
            },
            "ref": null,
            "rendered": 2,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "10",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              variant="body2"
>
              ...
</WithStyles(Typography)>,
            <WithStyles(Typography)
              className="style-bordered"
              onClick={[Function]}
              variant="button"
>
              10
</WithStyles(Typography)>,
          ],
          "className": "style-horizontal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "...",
              "variant": "body2",
            },
            "ref": null,
            "rendered": "...",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 10,
              "className": "style-bordered",
              "onClick": [Function],
              "variant": "button",
            },
            "ref": null,
            "rendered": 10,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="style-horizontal"
>
            <WithStyles(Typography)
                        className="style-bordered style-active"
                        onClick={[Function]}
                        variant="button"
            >
                        1
            </WithStyles(Typography)>
</div>,
          <div
            className="style-horizontal"
>
            <WithStyles(Typography)
                        className="style-bordered"
                        onClick={[Function]}
                        variant="button"
            >
                        2
            </WithStyles(Typography)>
</div>,
          <div
            className="style-horizontal"
>
            <WithStyles(Typography)
                        variant="body2"
            >
                        ...
            </WithStyles(Typography)>
            <WithStyles(Typography)
                        className="style-bordered"
                        onClick={[Function]}
                        variant="button"
            >
                        10
            </WithStyles(Typography)>
</div>,
        ],
        "className": "style-horizontal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <WithStyles(Typography)
                className="style-bordered style-active"
                onClick={[Function]}
                variant="button"
>
                1
</WithStyles(Typography)>,
            ],
            "className": "style-horizontal",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 1,
                "className": "style-bordered style-active",
                "onClick": [Function],
                "variant": "button",
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <WithStyles(Typography)
                className="style-bordered"
                onClick={[Function]}
                variant="button"
>
                2
</WithStyles(Typography)>,
            ],
            "className": "style-horizontal",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 2,
                "className": "style-bordered",
                "onClick": [Function],
                "variant": "button",
              },
              "ref": null,
              "rendered": 2,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "10",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                variant="body2"
>
                ...
</WithStyles(Typography)>,
              <WithStyles(Typography)
                className="style-bordered"
                onClick={[Function]}
                variant="button"
>
                10
</WithStyles(Typography)>,
            ],
            "className": "style-horizontal",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "...",
                "variant": "body2",
              },
              "ref": null,
              "rendered": "...",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 10,
                "className": "style-bordered",
                "onClick": [Function],
                "variant": "button",
              },
              "ref": null,
              "rendered": 10,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Pagination /> should render Pagination properly when pagesize is specified 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaginationBase
    classes={
        Object {
            "active": "style-active",
            "bordered": "style-bordered",
            "horizontal": "style-horizontal",
          }
    }
    count={145}
    onClick={[Function]}
    pageSize={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="style-horizontal"
>
          <WithStyles(Typography)
                    className="style-bordered style-active"
                    onClick={[Function]}
                    variant="button"
          >
                    1
          </WithStyles(Typography)>
</div>,
        <div
          className="style-horizontal"
>
          <WithStyles(Typography)
                    className="style-bordered"
                    onClick={[Function]}
                    variant="button"
          >
                    2
          </WithStyles(Typography)>
</div>,
        <div
          className="style-horizontal"
>
          <WithStyles(Typography)
                    variant="body2"
          >
                    ...
          </WithStyles(Typography)>
          <WithStyles(Typography)
                    className="style-bordered"
                    onClick={[Function]}
                    variant="button"
          >
                    49
          </WithStyles(Typography)>
</div>,
      ],
      "className": "style-horizontal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <WithStyles(Typography)
              className="style-bordered style-active"
              onClick={[Function]}
              variant="button"
>
              1
</WithStyles(Typography)>,
          ],
          "className": "style-horizontal",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 1,
              "className": "style-bordered style-active",
              "onClick": [Function],
              "variant": "button",
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <WithStyles(Typography)
              className="style-bordered"
              onClick={[Function]}
              variant="button"
>
              2
</WithStyles(Typography)>,
          ],
          "className": "style-horizontal",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 2,
              "className": "style-bordered",
              "onClick": [Function],
              "variant": "button",
            },
            "ref": null,
            "rendered": 2,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "49",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              variant="body2"
>
              ...
</WithStyles(Typography)>,
            <WithStyles(Typography)
              className="style-bordered"
              onClick={[Function]}
              variant="button"
>
              49
</WithStyles(Typography)>,
          ],
          "className": "style-horizontal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "...",
              "variant": "body2",
            },
            "ref": null,
            "rendered": "...",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 49,
              "className": "style-bordered",
              "onClick": [Function],
              "variant": "button",
            },
            "ref": null,
            "rendered": 49,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="style-horizontal"
>
            <WithStyles(Typography)
                        className="style-bordered style-active"
                        onClick={[Function]}
                        variant="button"
            >
                        1
            </WithStyles(Typography)>
</div>,
          <div
            className="style-horizontal"
>
            <WithStyles(Typography)
                        className="style-bordered"
                        onClick={[Function]}
                        variant="button"
            >
                        2
            </WithStyles(Typography)>
</div>,
          <div
            className="style-horizontal"
>
            <WithStyles(Typography)
                        variant="body2"
            >
                        ...
            </WithStyles(Typography)>
            <WithStyles(Typography)
                        className="style-bordered"
                        onClick={[Function]}
                        variant="button"
            >
                        49
            </WithStyles(Typography)>
</div>,
        ],
        "className": "style-horizontal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <WithStyles(Typography)
                className="style-bordered style-active"
                onClick={[Function]}
                variant="button"
>
                1
</WithStyles(Typography)>,
            ],
            "className": "style-horizontal",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 1,
                "className": "style-bordered style-active",
                "onClick": [Function],
                "variant": "button",
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <WithStyles(Typography)
                className="style-bordered"
                onClick={[Function]}
                variant="button"
>
                2
</WithStyles(Typography)>,
            ],
            "className": "style-horizontal",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 2,
                "className": "style-bordered",
                "onClick": [Function],
                "variant": "button",
              },
              "ref": null,
              "rendered": 2,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "49",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                variant="body2"
>
                ...
</WithStyles(Typography)>,
              <WithStyles(Typography)
                className="style-bordered"
                onClick={[Function]}
                variant="button"
>
                49
</WithStyles(Typography)>,
            ],
            "className": "style-horizontal",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "...",
                "variant": "body2",
              },
              "ref": null,
              "rendered": "...",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 49,
                "className": "style-bordered",
                "onClick": [Function],
                "variant": "button",
              },
              "ref": null,
              "rendered": 49,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Pagination /> should render Pagination properly when there is no items 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaginationBase
    classes={
        Object {
            "active": "style-active",
            "bordered": "style-bordered",
            "horizontal": "style-horizontal",
          }
    }
    count={0}
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Pagination /> should render Pagination properly when there is only one page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaginationBase
    classes={
        Object {
            "active": "style-active",
            "bordered": "style-bordered",
            "horizontal": "style-horizontal",
          }
    }
    count={29}
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="style-horizontal"
>
          <WithStyles(Typography)
                    className="style-bordered style-active"
                    onClick={[Function]}
                    variant="button"
          >
                    1
          </WithStyles(Typography)>
</div>,
        <div
          className="style-horizontal"
>
          <WithStyles(Typography)
                    className="style-bordered"
                    onClick={[Function]}
                    variant="button"
          >
                    2
          </WithStyles(Typography)>
</div>,
      ],
      "className": "style-horizontal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <WithStyles(Typography)
              className="style-bordered style-active"
              onClick={[Function]}
              variant="button"
>
              1
</WithStyles(Typography)>,
          ],
          "className": "style-horizontal",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 1,
              "className": "style-bordered style-active",
              "onClick": [Function],
              "variant": "button",
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <WithStyles(Typography)
              className="style-bordered"
              onClick={[Function]}
              variant="button"
>
              2
</WithStyles(Typography)>,
          ],
          "className": "style-horizontal",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 2,
              "className": "style-bordered",
              "onClick": [Function],
              "variant": "button",
            },
            "ref": null,
            "rendered": 2,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="style-horizontal"
>
            <WithStyles(Typography)
                        className="style-bordered style-active"
                        onClick={[Function]}
                        variant="button"
            >
                        1
            </WithStyles(Typography)>
</div>,
          <div
            className="style-horizontal"
>
            <WithStyles(Typography)
                        className="style-bordered"
                        onClick={[Function]}
                        variant="button"
            >
                        2
            </WithStyles(Typography)>
</div>,
        ],
        "className": "style-horizontal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <WithStyles(Typography)
                className="style-bordered style-active"
                onClick={[Function]}
                variant="button"
>
                1
</WithStyles(Typography)>,
            ],
            "className": "style-horizontal",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 1,
                "className": "style-bordered style-active",
                "onClick": [Function],
                "variant": "button",
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <WithStyles(Typography)
                className="style-bordered"
                onClick={[Function]}
                variant="button"
>
                2
</WithStyles(Typography)>,
            ],
            "className": "style-horizontal",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 2,
                "className": "style-bordered",
                "onClick": [Function],
                "variant": "button",
              },
              "ref": null,
              "rendered": 2,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
